pkg/controller/statefulset/stateful_set_control_test.go

```diff
@@ -1006,6 +1088,15 @@ func TestStatefulSetControlRollingUpdateWithMaxUnavailableInOrderedModeVerifyInv
 				}
 			}
 
+			for i := 0; i < totalPods; i++ {
+				if _, ok := tc.ordinalOfPodUnavailable[i]; ok {
+					// set to some time in the future
+					spc.setPodAvailable(set, i, time.Now().Add(time.Duration(2*tc.minReadySecs)*time.Second))
```

Comment:
If you're using time directly, it would make the test more clear if:

1. You set the time arbitrarily as a value above
2. You update updateStatefulSetAfterInvariantEstablished to take a `clock.Clock` (i.e. a fake)
3. You pass a fake clock set to the time in 1 into the method in the test case

I would do this after we get the e2e test working - I would just prefer to clean up test cases that need time injection as we go.